File: FrontEndCDAC/src/Customer/Data/productService.js
import axios from 'axios';

const API_URL = 'http://localhost:5454'; 

const productService = {
  getAllProducts: async () => {
    try {
      const response = await axios.get(`${API_URL}/products`);
      return response.data;
    } catch (error) {
      console.error('Error fetching all products:', error);
      return [];
    }
  },
  getProductsByCategory: async (category) => {
    try {
      const response = await axios.get(`${API_URL}/products/${category}`);
      return response.data;
    } catch (error) {
      console.error(`Error fetching products by category ${category}:`, error);
      return [];
    }
  }
};

export default productService;

File: cdac-proj/src/main/java/com/artgallery/cdacproj/service/ProductService.java
package com.artgallery.cdacproj.service;

import java.util.List;
import java.util.Optional;

import com.artgallery.cdacproj.model.Product;

public interface ProductService {

	List<Product> getAllProducts(); // Getting all products in the database

	List<Product> getProductsByCategory(String category);

<<<<<<< HEAD

=======
>>>>>>> 7c2f37acf0437f0ae7dab3513e64f2640044c213
	List<Product> getProductByArtistName(String artistName);

	void deleteById(int id);

	void addnewProduct(Product p);
<<<<<<< HEAD

	
	Optional<Product> findById(long id);




=======
>>>>>>> 7c2f37acf0437f0ae7dab3513e64f2640044c213


	Product getById(int id);

<<<<<<< HEAD
}


=======

	Product findById(Long productId);

}

>>>>>>> 7c2f37acf0437f0ae7dab3513e64f2640044c213
